source/mixin/network_mixin.py

```diff
@@ -0,0 +1,40 @@
+from PyQt4.QtCore import *
+from PyQt4.QtNetwork import *
+
+
+class NetworkMixin(QNetworkAccessManager):
+    request_url = str
+    result_connection = ""
+
+    @pyqtSlot()
+    def connectionFinished(self):
+        print(self.result_connection)
+
+    @pyqtSlot()
+    def connectionReadData(self):
+        self.result_connection += self.reply.readAll()
+
+    def connect(self):
+        print "connect it"
+        QObject.connect(
```

Comment:
@meomancer Also please read carefully https://github.com/inasafe/inasafe/wiki/Coding-Standards#qt-guidelines (and the rest of the coding guidelines in that document) - it should be the basis for all your QGIS Python plugin coding.